app/components/teacher/TeacherSetting/TeacherSettingRegion.tsx

```
@@ -1,60 +1,98 @@
   const [activeButtons, setActiveButtons] = useState<number[]>([]);
+  const [initialActive, setInitialActive] = useState<number[]>([]);
+
+  const { data, isLoading, error } = useGetTeacherSettingInfo({
+    name: teacherName,
+    phoneNumber: teacherPhone,
+  });
+
+  useEffect(() => {
+    if (data) {
+      const initial = buttonLabels.reduce<number[]>((acc, label, index) => {
+        return data.districts.includes(label) ? [...acc, index] : acc;
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)